### DISCUSSION TITLE
Is it possible to set window title bar in eframe?

### DISCUSSION THREAD

**Discussion by AkiaCode** [emilk/egui#781] [Q&A] [answered]
I want to use it, but I can't find it anywhere else

**Comment by NiceneNerd** [comment]
I second this question.

**Comment by AlexApps99** [comment, marked as answer]
AFAIK, you control it with the return value of [`epi::App::name()`](https://docs.rs/epi/0.14.0/epi/trait.App.html#tymethod.name)

**Reply by NiceneNerd** [reply]
That only seems to work once. I should have specified I need to be able to change it.

**Reply by AlexApps99** [reply]
In that case, I don't believe there is a way yet. You could add it in a PR, or alternatively, you could setup the windowing yourself (there are some examples in this repo showing how to do that) and manually change the title that way.

**Reply by emilk** [reply]
We should expose it via the `Frame` handle (which [should also be refactored](https://github.com/emilk/egui/issues/673)).

**Reply by NiceneNerd** [reply]
I decided to look at this, and my naive first guess based using `Frame::set_window_size` as a guide was to add a `window_title: Option<String>` field to `AppOutput`, until I noticed it has to be `Copy`. Not sure the best alternative route without delving myself into the larger refactor.

**Reply by emilk** [reply]
This is the right approach. `AppOutput` is currently `Copy`, but there is really no need for it to be so (as far as I can recall).

**Reply by NiceneNerd** [reply]
Cool, I may submit a PR later (looks like there are just two places where `AppOutput` would need to be cloned, but for a small optional string I expect that's no biggie)

**Reply by sdasda7777** [reply]
Is there any way to do this as of yet? 🤔

**Comment by emilk** [comment]
You can now change the name (and other stuff) of the OS Window using [`ViewportCommand`](https://docs.rs/egui/latest/egui/struct.Context.html#method.send_viewport_cmd)

**Reply by sdasda7777** [reply]
That's really nice, thank you for your reply!

However I do have to note, it appears that the update logic (which also refreshes the title) doesn't run when the app window is minimized, which I was hoping for. It makes perfect sense, of course, but is there any way to avoid this issue?
